The course prefix filter buttons style implemented

Urjikoo opened the pull request:
## Types of changes
<!--
  What types of changes does your code introduce? Put an `x` in all the boxes
  that apply:
-->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality
      to change)

## Checklist:
<!--
  Go over all the following points, and put an `x` in all the boxes that apply.
-->
- [X] I have run `eslint` on the code
- [X] I have added JSDoc for all of my code (where applicable)
- [ ] I have added tests to cover my changes.

## Priority:
- [X] Normal <!-- New piece of functionality -->
- [ ] High <!-- Critical bug requiring urgent review -->

## Related Issues:
<!--
  Add the issue number in below that this PR is intended to address.

  Also mention any issues that this PR is related to, and if possible,
  provide more information regarding the relationship to the issues in
  question (i.e does the PR fix the issue, is it designed to fix another bug
  that is similar to the issue etc.)
-->
Fixes #576 

<!--
  Attribution:
  PR Template adapted from: https://github.com/h5bp/html5-boilerplate/blob/master/.github/PULL_REQUEST_TEMPLATE.md
-->

Urjikoo committed:
The course prefix filter buttons style implemented
course prefix filter buttons background color was updated to show their state. When the button is clicked the background will be removed and goes to white but the font and the border shows the prefix color and background color will revert back to prefix color once it clicked again to show it's inactive.

jonseitz commented on src/client/components/pages/Schedule/SchedulePage.tsx:
```diff
@@ -216,9 +231,12 @@ const SchedulePage: FunctionComponent = () => {
     <>
-      <VerticalSpace>
+      <div aria-label="course filter buttons">
+        <span id="Course Filter Button">show/hide catalog prefixes</span>
```
Per the [HTML spec][mdnhtml], the id field shouldn't contain white space. I'd suggest kebab case (`id="course-filter-buttons"`) instead. 

[mdnhtml]: https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/id

jonseitz replied:
This text should be title-cased: `Show/Hide Catalog Prefixes`

jonseitz commented on src/client/components/pages/Schedule/SchedulePage.tsx:
```diff
@@ -216,9 +231,12 @@ const SchedulePage: FunctionComponent = () => {
         {prefixes.map((prefixObj) => (
           <PrefixButton
+            isPrefixActive={isPrefixActive}
+            aria-labelledby="Course Filter Button"
```
This should be updated to match the kebab-cased id suggested above

jonseitz commented on src/client/components/pages/Schedule/SchedulePage.tsx:
```diff
@@ -46,8 +46,23 @@ export interface PrefixState{
+const PrefixButton = styled(Button) <ButtonProps & { prefix: string, isPrefixActive: (prefix: string) => boolean; }>`
+ margin: 8px;
+ color: ${({ isPrefixActive, prefix }) => (isPrefixActive(prefix) ? 'black' : `${getCatPrefixColor(prefix)}`)};
+ border: ${({ isPrefixActive, prefix }) => (isPrefixActive(prefix) ? '' : `1px solid ${getCatPrefixColor(prefix)}`)};
```
This `border` value is causing a minor bug where the size of the overall box changes when button are clicked. I've played around with a couple of different solutions, and I think we can either:

1. Don't change the `border`, `color`, and `background-color`, but instead change the `opacity` of the button as a whole to match what's in the table.
2. Add the border to all states on the buttons (including the `:hover` and `:active` pseudo-classes

I'm personally leaning toward 1, but I think either approach is fine.

jonseitz commented on src/client/components/pages/Schedule/SchedulePage.tsx:
```diff
@@ -216,9 +231,12 @@ const SchedulePage: FunctionComponent = () => {
+        <span id="Course Filter Button">show/hide catalog prefixes</span>
         {prefixes.map((prefixObj) => (
           <PrefixButton
+            isPrefixActive={isPrefixActive}
```
Rather than passing `isPrefixActive` as a function, I'd suggest making it a boolean prop and just calling it with `prefix` here as use time. We can also change the name to just `isActive` to avoid confusion with the function defined above.

```tsx
  isActive={isPrefixActive(prefixObj.prefix)}
``` 

I don't think this makes too much of a different technically, but it does make the component definition easier when you don't need as many embedded calls to `isPrefixActive(prefix)`...

jonseitz reviewed (changes requested):
This is looking really good. I have just a handful of style notes and a few things related to conventions we've used elsewhere in the project or in Mark One.

Urjikoo committed:
Styling for dropdown filter and button was implemented.
MenuFlex was added and imported Markone theme for spacing for dropdown menu. Prefix button filter buttons' styling was updatedadjusting the opacity of the button depending on the state and font and border stylings were removed as they no longer needed to show the state of the button.

Urjikoo committed:
prefix button styling function was updated.
prefix button state was passed as a boolean instead of a function to reduce how many times we're calling the function.

jonseitz reviewed (approved):
Pending the tests re-running successfully, this should be all set
